### PR TITLE
Fix Shell.mv

### LINKED ISSUE #2293: `Shell.mv` doesn't move correctly
### Description

`Shell.mv` should behave like `mv` in shell, as described in documentation.

### Repro steps
https://github.com/fangyi-zhou/fake-mv-bug
- Create a file `1`
- `Shell.mv "1" "2"`

```Script reported an error:
-> BuildFailedException: Target 'All' failed.
-> One or more errors occurred. (Could not find file '/Users/fangyi/fake-mv-bug/2'.)
-> FileNotFoundException: Could not find file '/Users/fangyi/fake-mv-bug/2'.
```

### Expected behavior
Rename `1` to `2`
### Actual behavior
`FileNotFoundException`
### Known workarounds
`Shell.rename`
### Related information 
FAKE 5.13

### PR DESCRIPTION
**Pull request by fangyi-zhou**
### Description

Fixes #2293 
## TODO

- [ ] New (API-)documentation for new features exist (Note: API-docs are enough, additional docs are in `help/markdown`)
- [x] unit or integration test exists (or short reasoning why it doesn't make sense)
- [x] boy scout rule: "leave the code behind in a better state than you found it" (fix warnings, obsolete members or code-style in the places you worked in)     
- [ ] (if new module) the module has been linked from the "Modules" menu, edit `help/templates/template.cshtml`, linking to the API-reference is fine.
- [ ] (if new module) the module is in the correct namespace
- [ ] (if new module) the module is added to Fake.sln (`dotnet sln Fake.sln add src/app/Fake.*/Fake.*.fsproj`)
- [ ] Fake 5 [API guideline](https://fake.build/contributing.html#API-Design) is honored

### CODE REVIEW

**Comment by fangyi-zhou** [comment]
Moving for directories isn't working...

**Comment by matthid** [comment]
a good start, thanks for taking a look at this. 

Just FYI some tests are still failing, what I use is something like `dotnet run --project src/test/Fake.Core.IntegrationTests/ -- --run "Fake.IO.FileIntegraionTests/Shell.mv works as expected when src is a dir dst is a dir - #2293"` to run them locally one-by one and have the dotnet tooling rebuild everything if needed.

File: src/app/Fake.IO.FileSystem/FileSystemInfo.fs [hunk 1]
```diff
@@ -25,3 +25,9 @@ module FileSystemInfo =
         | :? FileInfo as file -> File(file)
         | :? DirectoryInfo as dir -> Directory(dir, dir.EnumerateFileSystemInfos())
         | _ -> failwith "No file or directory given."
+
+    let moveTo (fileSysInfo: FileSystemInfo) dest =
```
Comment (matthid) [inline src/app/Fake.IO.FileSystem/FileSystemInfo.fs]: We should either have docs or mark this `internal` (which is what I probably would go with for now).
If we want to add this public API we should briefly discuss the order of the arguments.

What I mean: We should decide if we expect this API to be used like `FileSystemInfo.moveTo src dst` or like

```fsharp
src
|> FileSystemInfo.moveTo dst
```
in which case we have to re-order the arguments.

File: src/app/Fake.IO.FileSystem/Shell.fs [hunk 2]
```diff
@@ -337,6 +339,14 @@ module Shell =
             f.MoveTo(targetName) |> ignore
         | FileSystemInfo.Directory _ -> () //TODO: logVerbosefn "Ignoring %s, because it is a directory." fileName
 
+    let private moveDir target fileName =
+        let fi = FileSystemInfo.ofPath fileName
+        match fi with
+        | FileSystemInfo.Directory (d, _) ->
+            let targetName = target @@ fi.Name
+            d.MoveTo(targetName) |> ignore
```
Comment (matthid) [inline src/app/Fake.IO.FileSystem/Shell.fs]: can we write `ignore<DirectoryInfo>` here (or whatever `MoveTo` returns here)?

File: src/app/Fake.IO.FileSystem/Shell.fs [hunk 3]
```diff
@@ -646,4 +656,20 @@ module Shell =
     /// Like "mv" in a shell. Moves/renames a file
     /// <param name="src">The source</param>
     /// <param name="dest">The destination</param>
-    let mv src dest = moveFile dest src
+    let mv src dest =
+        let fi_src = FileSystemInfo.ofPath src
+        let fi_dest = FileSystemInfo.ofPath dest
+        if not fi_dest.Exists then rename dest src
+        else begin
```
Comment (matthid) [inline src/app/Fake.IO.FileSystem/Shell.fs]: I think `begin` and `end` can be removed

**Comment by matthid** [comment]
Thanks!

File: src/app/Fake.IO.FileSystem/Shell.fs (hunk 2 above)
Comment (fangyi-zhou) [reply src/app/Fake.IO.FileSystem/Shell.fs]: The method returns `void`, so I guess we don't need to `ignore`?

**Comment by fangyi-zhou** [comment]
Please revert... that code was not compiling... Sorry

**Comment by matthid** [comment]
It's ok, will fix and start the release process :)
I have to edit the release notes anyway and wanted to try some refactoring I wasn't sure if it would work in the review ;)

**Comment by matthid** [comment]
Honestly, I didn't expect `mv` to be such a nasty little function :D

File: src/app/Fake.IO.FileSystem/Shell.fs (hunk 2 above)
Comment (matthid) [reply src/app/Fake.IO.FileSystem/Shell.fs]: Indeed

**Pull request merged**